packages/iov-core/src/signingservercore.ts

```diff
@@ -49,10 +77,12 @@ export class SigningServerCore {
         throw new Error("More than one wallets contain the identity to sign this transaction");
     }
 
-    // TODO: ask user for permission
-
-    const response = await this.signer.signAndPost(transaction, walletId);
-    return response.transactionId;
+    if (this.authorizeSignAndPost(reason, transaction)) {
```

Comment:
Isn't this a promise??
Don't you want: `if (await this.authorizeSignAndPost(reason, transaction))`...

Reply:
Well spotted, this is a horrible bug!

Will add a test for the other case that finds this bug